src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -1124,18 +1123,12 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     {
                         if (HWIntrinsicInfo::IsExplicitMaskedOperation(intrin.id))
                         {
-                            assert((targetReg == op1Reg) || (targetReg != op1Reg));
```

Comment:
these are OK to have.

Reply:
These ones are hit by the new hwintrinsic tests:

```
------------------- {'JitStressRegs': '1'} -------------------
Errors:

Assert failure(PID 366410 [0x0005974a], Thread: 366410 [0x5974a]): Assertion failed '(targetReg == op1Reg) || (targetReg != op3Reg)' in 'JIT.HardwareIntrinsics.Arm._Sve.SimpleTernaryOpTest__Sve_ConditionalExtractAfterLastActiveElementAndReplicate_float:RunSameClassFldScenario():this' during 'Generate code' (IL size 92; hash 0xa42df59e; FullOpts)

    File: /home/alahay01/dotnet/runtime_sve_api/src/coreclr/jit/hwintrinsiccodegenarm64.cpp:1118
    Image: /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/linux.arm64.Checked/Tests/Core_Root/corerun
```

Reply:
We need some kind of assert validating the `mov` is valid.

Since we're doing `mov targetReg, op2Reg`, then when `targetReg == op1Reg` and `op1Reg != op2Reg`, then the we would overwrite `op1` and produce in incorrect results.

So, therefore, we expect that `targetReg == op2Reg` so no `mov` is emitted.

The assert therefore needs to be something like `assert((targetReg == op2Reg) || ((targetReg != op1Reg) && (targetReg != op3Reg))`. Which validates `no move` or `no overwrite`

Reply:
All paths need some kind of similar assert validating `no move` or `no overwrite, if a move is emitted`

Reply:
> All paths need some kind of similar assert validating `no move` or `no overwrite, if a move is emitted`

Agree.

Reply:
Done replaced this assert as suggested, and the next one with a similar assert

Reply:
confirmed stress tests passing.